engine/storage/src/main/java/org/apache/cloudstack/storage/image/BaseImageStoreDriverImpl.java

```diff
@@ -391,12 +391,16 @@ public void copyAsync(DataObject srcdata, DataObject destData, AsyncCompletionCa
     private Answer sendToLeastBusyEndpoint(List<EndPoint> eps, CopyCommand cmd) {
         Answer answer = null;
         EndPoint endPoint = null;
-        Long epId = ssvmWithLeastMigrateJobs();
+        List<Long> epIds = ssvmWithLeastMigrateJobs();
+        Long epId = null;
+        if (!epIds.isEmpty()) {
+            epId = epIds.get(0);
+        }
         if (epId == null) {
```

Comment:
@Pearl1594 seems `epId` is no longer required for the check here. Can check with empty `epIds` ?

Reply:
@sureshanaparti is this a remark to style? if epIds is empty epId is null so the check is needed in some way.

Reply:
@DaanHoogland the last method _ssvmWithLeastMigrateJobs()_ returns null if it fails to pick ssvm and the null check seems to be valid. The same updated method now returns an empty list if fails. So list with empty check is fine, 'epId' may not be required.